File: src/326. Power of Three.js
/**
 * @param {number} n
 * @return {boolean}
 */

// function isPowerOfThree(n) {
//   if (n === 1) return true
//   return n > 1 ? isPowerOfThree(n / 3) : false
// }

export function isPowerOfThree(n) {
  return n > 0 && 1162261467 % n == 0
}

File: src/326. Power of Three.cpp
class Solution {
public:
    // bool isPowerOfThree(int n) {
    //     if (n == 1) return true;
    //     if (n % 3 || n < 1) return false;
    //     return isPowerOfThree(n / 3);
    // }

    bool isPowerOfThree(int n) {
        return n > 0 && 1162261467 % n == 0;
    }
};